factories/comments.rb

```diff
@@ -10,6 +10,14 @@
     pseud
   end
 
+  factory :guest_comment, class: Comment do
```

Comment:
hahaha sob, I typed up a comment asking why we added this when it's the same as the one above it... but now I see the one above it is some hybrid logged-in/guest comment monstrosity that should never occur in the wilds of the Archive, so that's cool and wonderful our comment tests have been using that. 🤦

Reply:
On top of being simultaneously registered and non-registered, the comment factory defaults to having the commentable being a work, which is not a great model of what a real comment looks like, because [all comments are on chapters rather than on works](https://github.com/otwcode/otwarchive/blob/3b2cf664ea303b1066f7372b6930c9b51cef64b0/app/models/comment.rb#L367-L369).

(Incidentally, I suspect the work/chapter issue is why [this issue with comments being included in the stats](https://github.com/otwcode/otwarchive/blob/3b2cf664ea303b1066f7372b6930c9b51cef64b0/features/step_definitions/work_search_steps.rb#L137-L146) occurred. If the comments were generated with `FactoryBot.create(:comment, commentable: work)`, they wouldn't show up in the stats because the stats only include comments on chapters.)

Reply:
Our Slack emoji for sobbing blood would be handy right now.

I'll make an issue for fixing up the comment factory.

Reply:
[AO3-5999](https://otwarchive.atlassian.net/browse/AO3-5999) now exists for updating the existing factories, but it would be great if you could update the commentable on the new factory to be a chapter, @CristinaRO.

Reply:
#3870 added a factory for guest comments and was merged, you can use that now.